signature/src/keypair.rs

```diff
@@ -0,0 +1,64 @@
+pub trait KeyPair: core::fmt::Debug + Sized {
+    fn get_public_key_encoded(&mut self) -> String
+    where
+        Self: Sized;
+    fn get_private_key_encoded(&mut self) -> String
+    where
+        Self: Sized;
+    fn get_public_key(&self) -> ed25519_zebra::VerificationKey
+    where
+        Self: Sized;
+    fn get_private_key(&self) -> ed25519_zebra::SigningKey
+    where
+        Self: Sized;
+}
+
+#[derive(Debug)]
+pub struct SSIKeyPair {
+    pub(crate) public_key_encoded: Option<String>,
+    pub(crate) private_key_encoded: Option<String>,
+    pub(crate) private_key: ed25519_zebra::SigningKey,
+    pub(crate) public_key: ed25519_zebra::VerificationKey,
+}
+
+impl SSIKeyPair {
+    fn new() -> Self {
+        let sk = ed25519_zebra::SigningKey::new(rand::thread_rng());
+        let vk = ed25519_zebra::VerificationKey::from(&sk);
+        return Self {
+            public_key_encoded: None,
```

Comment:
Why not remove these properties?

Reply:
Lazy-load implemented, once someone calls get_public_key_encoded, it converts the key to a string and stores it in the property.